[deps][llm] vLLM 0.13.0

eicherseiji opened the pull request:
## Summary
- Workaround for https://github.com/vllm-project/vllm/issues/30818
- Remove tokenizer `None` due to https://github.com/vllm-project/vllm/pull/29693/files#diff-31e6bd0df09a47b5587701203d558701ac46e4f85bf7db83632da9990eaef198L1627
- need to bump `anyio`
```
  ╰─▶ Because all versions of mcp depend on anyio==3.7.1 and vllm==0.13.0
      depends on mcp, we can conclude that vllm==0.13.0 depends on
      anyio==3.7.1. (1)
...
      and mcp<=1.1.0 depends on anyio>=4.6, we can conclude that mcp<1.1.1
      depends on anyio>=4.6.
      And because mcp>=1.1.1 depends on anyio>=4.5 and vllm==0.13.0 depends on
      mcp, we can conclude that vllm==0.13.0 depends on anyio>=4.5.
      And because we know from (1) that vllm==0.13.0 depends on anyio==3.7.1,
      we can conclude that vllm[audio]==0.13.0 cannot be used.
      And because only vllm[audio]<=0.13.0 is available and you require
      vllm[audio]>=0.13.0, we can conclude that your requirements are
      unsatisfiable.

      hint: Pre-releases are available for `mcp` in the requested range (e.g.,
      1.3.0rc1), but pre-releases weren't enabled (try: `--prerelease=allow`)
```

  - so need to bump `jupyterlab` and others
 ```
python/requirements/cloud-requirements.txt: jupyterlab==3.6.1 → jupyterlab>=4.0.0
python/requirements/ml/tune-test-requirements.txt: jupyterlab==3.6.1 → jupyterlab>=4.0.0
python/requirements/test-requirements.txt: tornado==6.1 → tornado>=6.2.0
ci/raydepsets/configs/rayllm.depsets.yaml: Added --upgrade-package anyio to common_settings
python/requirements_compiled.txt: Regenerated (now has jupyterlab 4.5.1, jupyter-server 2.10.0, tornado 6.5.4)
```

eicherseiji committed:
vLLM 0.13.0

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Bump python-platform

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Temporarily add no build isolation to depsets

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
More temp workarounds to get the non-pypi RC loaded in CI

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Try not updating CPU lockfiles for now

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Use pre-built wheel

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Temp update release tests

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Don't update extra libs scripts yet

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Roll back CPU lock files for now

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Remove None tokenizer arg in udf

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Fix MediaWithBytes reducer

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Remove tokenizer from test_batch_vllm.py

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Prep for pypi

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Bump jupyterlab, tornado, anyio

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

aslonnie reviewed (commented):
try sync master? python 3.9 is removed already.

eicherseiji committed:
Merge branch 'master' into vllm-0.13.0

eicherseiji committed:
Try merge master

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

elliot-barn commented on python/deplocks/llm/rayllm_subset_py311_cu128.lock:
```diff
@@ -1,5 +1,9 @@
+--find-links https://wheels.vllm.ai/7d80c73d4277187d0468f15a22bba959ce853261/cu129/vllm/
+--find-links https://wheels.vllm.ai/7d80c73d4277187d0468f15a22bba959ce853261/cu129/vllm/
+--find-links https://wheels.vllm.ai/7d80c73d4277187d0468f15a22bba959ce853261/cu129/vllm/
+--find-links https://wheels.vllm.ai/7d80c73d4277187d0468f15a22bba959ce853261/cu129/vllm/
```
I'll fix this in another PR

elliot-barn commented:
LGTM! Would be good to test a notebook in a workspace. I cloned a workspace from a release test job, but its using jupyterlab 4.5 [here](https://console.anyscale-staging.com/o/anyscale-internal/jobs/prodjob_ehwx1qb2111nsbd4r6mht66fzw)

eicherseiji commented:
> Also just run ci/ci.sh compile_pip_dependencies && bazel run //ci/raydepsets:raydepsets -- build --all-configs 1 more time for the dependency compilation to pass the dependency tests

@elliot-barn got the error:
```
  ╰─▶ Because you require jupyter-server-terminals==0.4.4 and jupyter-server-terminals==0.5.3, we can conclude that your requirements are unsatisfiable.
```

But `/home/ray/default/work/ray/docker/base-extra/requirements.in` requires this `jupyter_server_terminals` version due to a `terminado` constraint:
```
# Pinning jupyter_server_terminals==0.4.4 , the higher version will break the
# webterminal when using an older version of terminado.
jupyter_server_terminals==0.4.4
```

elliot-barn committed:
updating dependencies

Signed-off-by: elliot-barn <elliot.barnwell@anyscale.com>

elliot-barn commented on docker/base-extra/requirements.in:
```diff
@@ -6,7 +6,7 @@ grpcio-tools
 # Pinning jupyter_server_terminals==0.4.4 , the higher version will break the
 # webterminal when using an older version of terminado.
-jupyter_server_terminals==0.4.4
+jupyter_server_terminals
```
Yes please and thanks

eicherseiji committed:
Remove comment

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Merge branch 'vllm-0.13.0' of https://github.com/eicherseiji/ray into vllm-0.13.0

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Update task kwarg

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Fix task_type

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Merge branch 'master' of https://github.com/ray-project/ray into vllm-0.13.0

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Recompile deps

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Use requirements from CI

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Recompile

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Update new tests, copy engine_kwargs dict before mutating

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Resolve conflicts

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Change task -> task_type

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Fix task type override logic

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

jeffreywang-anyscale commented on python/ray/llm/_internal/batch/stages/vllm_engine_stage.py:
```diff
@@ -638,6 +655,11 @@ def normalize_engine_kwargs(
+        # Set task_type for the wrapper to use, but only if not already set.
+        # The processor may have already set this value.
+        if "task_type" not in engine_kwargs:
+            engine_kwargs["task_type"] = self.task_type
```
nit: Under what conditions is this code path activated? Doesn’t the processor always set `task_type` via `validate_task_type`?

eicherseiji replied:
Nice catch, I think it can be removed

jeffreywang88 reviewed (approved):
Ray Data LLM changes look good to me in general. Leaving a nit.

eicherseiji committed:
Task type set by processor

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Update test

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

eicherseiji committed:
Merge branch 'master' into vllm-0.13.0

eicherseiji committed:
Remove unnecessary anyio upgrade workaround

The requirements_compiled.txt already has anyio==4.12.0, which satisfies
mcp's anyio>=4.5 requirement. The --upgrade-package anyio flag is not needed.

Signed-off-by: Seiji Eicher <seiji@anyscale.com>

aslonnie reviewed (approved):
beautiful!

<div><img src="https://media2.giphy.com/media/p8GJOXwSNzQPu/giphy.gif?cid=5a38a5a2mqokgzcwbz8cegfgp2slmyxvo5irgkc38o323gwq&amp;ep=v1_gifs_search&amp;rid=giphy.gif&amp;ct=g" style="border:0;height:300px;width:300px"/><br/>via <a href="https://giphy.com/gifs/kill-me-now-p8GJOXwSNzQPu">GIPHY</a></div>

kouroshHakha reviewed (approved):
LGTM
